crates/wasi-http/src/proxy.rs

```diff
@@ -47,18 +47,24 @@ pub mod sync {
         async: false,
         with: {
             "wasi:http": bindings::http, // http is in this crate
-            "wasi:io": wasmtime_wasi::bindings::sync_io, // io is sync
+            "wasi:io": wasmtime_wasi::bindings::sync, // io is sync
             "wasi": wasmtime_wasi::bindings, // everything else
         },
     });
 
     pub fn add_to_linker<T>(l: &mut wasmtime::component::Linker<T>) -> anyhow::Result<()>
     where
-        T: WasiHttpView + wasmtime_wasi::WasiView + bindings::http::types::Host,
+        T: WasiHttpView + wasmtime_wasi::WasiView,
     {
-        // TODO: this shouldn't be required, but the adapter unconditionally pulls in all of these
-        // dependencies.
-        wasmtime_wasi::command::sync::add_to_linker(l)?;
+        wasmtime_wasi::bindings::clocks::wall_clock::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::clocks::monotonic_clock::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::sync::io::poll::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::sync::io::streams::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::io::error::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::cli::stdin::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::cli::stdout::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::cli::stderr::add_to_linker(l, |t| t)?;
+        wasmtime_wasi::bindings::random::random::add_to_linker(l, |t| t)?;
```

Comment:
The main reason is that the auto-generated functions all take a closure that's `fn(&mut T) -> &mut U` but that's switched here to be `T: WasiView`. Otherwise though, you're right, and it's something I've wanted to clean up for awhile.